### ISSUE TITLE
Conflict with Divi Builder API and Woocommerce Membership plugin update

### ISSUE THREAD

**Issue by jtrimm007** [woocommerce/woocommerce#20159]
When trying to update the Woocommerce Membership plugin, it throughs an error with the Divi Builder. Please view the image below for more details. 

Steps to replicate:
1. Go to update plugins
2. click update Woocommerce Membership plugin

Screenshots:
![image](https://user-images.githubusercontent.com/16611773/40328300-e5dca8c6-5d13-11e8-8836-1be655850a1d.png)

**WordPress Environment**
<details>
`
### WordPress Environment ###

Home URL: https://stantontradingpartners.com
Site URL: https://stantontradingpartners.com
WC Version: 3.3.5
Log Directory Writable: ✔
WP Version: 4.9.6
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.33
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.54.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.39
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.5
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 1.90MB
Database Data Size: 1.66MB
Database Index Size: 0.24MB
wp_woocommerce_sessions: Data: 0.01MB + Index: 0.00MB
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_order_items: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_order_itemmeta: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_commentmeta: Data: 0.00MB + Index: 0.01MB
wp_comments: Data: 0.00MB + Index: 0.01MB
wp_et_bloom_stats: Data: 0.00MB + Index: 0.00MB
wp_failed_jobs: Data: 0.00MB + Index: 0.00MB
wp_links: Data: 0.00MB + Index: 0.00MB
wp_mailchimp_carts: Data: 0.00MB + Index: 0.00MB
wp_options: Data: 0.77MB + Index: 0.06MB
wp_postmeta: Data: 0.17MB + Index: 0.07MB
wp_posts: Data: 0.67MB + Index: 0.04MB
wp_queue: Data: 0.03MB + Index: 0.00MB
wp_termmeta: Data: 0.00MB + Index: 0.01MB
wp_terms: Data: 0.00MB + Index: 0.01MB
wp_term_relationships: Data: 0.00MB + Index: 0.01MB
wp_term_taxonomy: Data: 0.00MB + Index: 0.00MB
wp_usermeta: Data: 0.01MB + Index: 0.01MB
wp_users: Data: 0.00MB + Index: 0.01MB
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB

### Post Type Counts ###

attachment: 38
custom_css: 1
customize_changeset: 7
et_pb_layout: 5
jetpack_migration: 2
jp_img_sitemap: 1
jp_sitemap: 1
jp_sitemap_master: 1
nav_menu_item: 12
page: 21
post: 3
product: 13
revision: 147
scheduled-action: 2
shop_order: 8
wc_membership_plan: 7
wc_user_membership: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (10) ###

Akismet Anti-Spam: by Automattic – 4.0.3
Bloom: by Elegant Themes – 1.2.25
Hello Dolly: by Matt Mullenweg – 1.6
Jetpack by WordPress.com: by Automattic – 6.1
Monarch Plugin: by Elegant Themes – 1.3.27
WooCommerce PayPal Express Checkout Gateway: by WooCommerce – 1.5.4
WooCommerce Stripe Gateway: by WooCommerce – 4.1.3
WooCommerce Memberships: by SkyVerge – 1.10.1 – 1.10.3 is available
WooCommerce Services: by Automattic – 1.13.3
WooCommerce: by Automattic – 3.3.5

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #39 - /shop/
Cart: #40 - /cart/
Checkout: #41 - /checkout/
My account: #42 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Divi
Version: 3.3.1
Author URL: http://www.elegantthemes.com
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
I've seen this before. I suspect Elegant Themes have used a similar upgrade mechanism to our extensions and it's breaking that process. If you disable the elegant themes product temporarily you should be able to update that way for now (or update manually uploading the new version via FTP).

This is not a core issue because it's extensions, but is something the extension team may want to look into as it affects extensions on the marketplace. cc @bor0 @avivapinchas 

Please post on the helpdesk in the meantime so someone can look at the account/install https://woocommerce.com/contact-us/

Thanks!

**Comment by jtrimm007** [comment]
Thank you for your timely reply. I will test it and let you know the outcome.

**Comment by mattyza** [comment]
I looked into this a little while ago. Here's what I found:

I did some further digging, and can reproduce this on my test store.

The issue stems from Elegant Themes’ irregular use of a `gettext` filter.

Divi checks for updates. If no updates are available, or no license key is authenticated for Divi, Divi hard returns the string of text in the original post, as a filter on `gettext`. Unfortunately, this particular filter doesn’t share any kind of context, so Divi is unable to make their filter code aware of whether or not the system is trying to update Divi itself, one of Divi’s companion plugins, or any other plugin.

The line they’re filtering on in WordPress core is https://github.com/WordPress/WordPress/blob/aaf99e691391cfceb004d848450dbbf3344b1bee/wp-admin/includes/update.php#L409

A solution to this, for Divi, would be to handle their authentication check in context, like WooCommerce and other providers do. We know that we’re dealing with a specific plugin before we do any checks, and can bounce out of the check if it’s not one of our plugins, or if it’s an unauthenticated update check. This is done with filters on `plugin_info_check` and others in the `WP_Upgrader` class.

cc @mikejolley @avivapinchas @bor0
